frame/composable-support/src/validation.rs

```diff
@@ -144,43 +233,62 @@ mod test {
 		b: u32,
 	}
 
-	impl Validate<ValidARange> for X {
-		fn validate(self) -> Result<X, &'static str> {
-			if self.a > 10 {
+	impl Validate<X, ValidARange> for ValidARange {
+		fn validate(extrinsic_input: X) -> Result<X, &'static str> {
+			if extrinsic_input.a > 10 {
 				Err("Out of range")
 			} else {
-				Ok(self)
+				Ok(extrinsic_input)
 			}
 		}
 	}
 
-	impl Validate<ValidBRange> for X {
-		fn validate(self) -> Result<X, &'static str> {
-			if self.b > 10 {
+	impl Validate<X, ValidBRange> for ValidBRange {
+		fn validate(extrinsic_input: X) -> Result<X, &'static str> {
+			if extrinsic_input.b > 10 {
 				Err("Out of range")
 			} else {
-				Ok(self)
+				Ok(extrinsic_input)
 			}
 		}
 	}
 
-	#[test]
-	fn nested_validator() {
-		let valid = X { a: 10, b: 0xCAFEBABE };
+	// #[test]
```

Comment:
is possible to restore nested?

Reply:
:+1:

Reply:
Nested restored here 7ff004b29650795dc5fbb23fb96035061d07ee8c